js/ui/codingManager.js

```diff
@@ -29,239 +46,525 @@ function _isBuilderSpeedwagon(window) {
+    Properties: {
+        window: GObject.ParamSpec.object('window',
+                                         '',
+                                         '',
+                                         GObject.ParamFlags.READWRITE |
+                                         GObject.ParamFlags.CONSTRUCT_ONLY,
+                                         Meta.Window),
+        builder_window: GObject.ParamSpec.object('builder-window',
+                                                 '',
+                                                 '',
+                                                 GObject.ParamFlags.READWRITE,
+                                                 Meta.Window)
+    },
+    Signals: {
+        'clicked': {}
+    },
+
+    _init: function(params) {
+        this.parent(params);
+
+        let rect = this.window.get_frame_rect();
+
+        // The button will be auto-added to the manager. Note that in order to
+        // remove this button, you will need to call eject() from outside
+        // this class or use removeChrome from within it.
+        this._button = _createViewSourceButtonInRectCorner(rect);
+
+        // Do required setup to make this button interactive, by allowing
+        // it to be focused, connecting to the button-press-event signal
+        // and adding chrome to the toplevel layoutManager.
+        this._button.reactive = true;
+        this._button.can_focus = true;
+        this._button.track_hover = true;
+        this._button.connect('button-press-event', Lang.bind(this, function() {
+            this.emit('clicked');
+        }));
+        Main.layoutManager.addChrome(this._button);
+
+        // Connect to signals on the window to determine when to move
+        // hide, and show the button. Note that WindowTrackingButton is
+        // constructed with the primary app window and we listen for signals
+        // on that. This is because of the assumption that both the app
+        // window and builder window are completely synchronized.
+        this._positionChangedId = this.window.connect(
+           'position-changed',  Lang.bind(this, this._updatePosition)
+        );
+        this._sizeChangedId = this.window.connect(
+            'size-changed', Lang.bind(this, this._updatePosition)
+        );
+
+        this._windowsRestackedId = Main.overview.connect(
+            'windows-restacked', Lang.bind(this, this._showIfWindowVisible)
+        );
+        this._overviewHidingId = Main.overview.connect(
+            'hiding', Lang.bind(this, this._showIfWindowVisible)
+        );
+        this._overviewShowingId = Main.overview.connect(
+            'showing', Lang.bind(this, this._hide)
+        );
+        this._windowMinimizedId = global.window_manager.connect(
+            'minimize', Lang.bind(this, this._hide)
+        );
+        this._windowUnminimizedId = global.window_manager.connect(
+            'unminimize', Lang.bind(this, this._show)
+        );
+    },
+
+    eject: function() {
```

Comment:
Nitpick: it's a bit odd to have this function named like this. Perhaps `remove`, or `destroy`?